File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/OccupyLocationUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.OccupyLocationCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.LocationResult;
import reactor.core.publisher.Mono;

public interface OccupyLocationUseCase {

    Mono<LocationResult> occupy(OccupyLocationCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/OccupyLocationUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.OccupyLocationCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.LocationResult
import reactor.core.publisher.Mono

interface OccupyLocationUseCase {
    fun occupy(command: OccupyLocationCommand): Mono<LocationResult>
}
